lucene/core/src/test/org/apache/lucene/search/TestConjunctionDISI.java

```diff
@@ -391,4 +391,20 @@ public void testCollapseSubConjunctionDISIs() throws IOException {
   public void testCollapseSubConjunctionScorers() throws IOException {
     testCollapseSubConjunctions(true);
   }
+
+  public void testIllegalAdvancementOfSubIteratorsTripsAssertion() throws IOException {
+    int maxDoc = 100;
+    final int numIterators = TestUtil.nextInt(random(), 2, 5);
+    FixedBitSet set = randomSet(maxDoc);
+
+    DocIdSetIterator[] iterators = new DocIdSetIterator[numIterators];
+    for (int i = 0; i < iterators.length; ++i) {
+      iterators[i] = new BitDocIdSet(set).iterator();
+    }
+    final DocIdSetIterator conjunction = ConjunctionDISI.intersectIterators(Arrays.asList(iterators));
+    int idx = TestUtil.nextInt(random() , 0, iterators.length-1);
+    iterators[idx].nextDoc(); // illegally advancing one of the sub-iterators outside of the conjunction iterator
+    AssertionError ex = expectThrows(AssertionError.class, () -> conjunction.nextDoc());
+    assertEquals(ex.getMessage(), "Sub-iterators of ConjunctionDISI are not the same document!");
```

Comment:
can you revert the two arguments to assertEquals? it expects the expected value first and the actual value second. While swapping arguments doesn't cause major issues, it creates confusing error messages when the assertion fails.

Reply:
Thanks @jpountz, my mistake, addressed in bf9a3e8